rct229/rules/section21/section21rule9.py

```diff
@@ -0,0 +1,83 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.assertions import getattr_
+from rct229.schema.config import ureg
+from rct229.utils.jsonpath_utils import find_all
+
+APPLICABLE_SYS_TYPES = [
+    "SYS-1",
+    "SYS-5",
+    "SYS-7",
+    "SYS-11.2",
+    "SYS-12",
+    "SYS-1A",
+    "SYS-7A",
+    "SYS-11.2A",
+    "SYS-12A",
+]
+PUMP_POWER_PER_FLOW_RATE = 19.0 * ureg("W/gpm")
+
+
+class Section21Rule9(RuleDefinitionListIndexedBase):
+    """Rule 9 of ASHRAE 90.1-2019 Appendix G Section 21 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section21Rule9, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section21Rule9.HeatingFluidLoopRule(),
+            index_rmr="baseline",
+            id="21-9",
+            description="When baseline building includes boilers, Hot Water Pump Power = 19W/gpm.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # FIXME: replace with baseline_system_types = get_baseline_system_types(rmi_b) when get_baseline_system_types
+        #  is ready.
+        baseline_system_types = {
+            "SYS-7": ["hvac_sys_7"],
+            "SYS-12": ["hvac_sys_12"],
+        }
+        # if any system type found in the APPLICABLE_SYS_TYPES then return applicable.
+        return any(
+            [key in APPLICABLE_SYS_TYPES for key in baseline_system_types.keys()]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        boilers = find_all("$.boilers[*]", rmi_b)
+        loop_boiler_dict = {}
```

Comment:
The RDS is overly complicated here. It creates this dictionary and then only uses its keys.  It would be more straight forward to simply create a list of boiler loop ids.
```
boiler_loop_ids = find_all("boilers[*].loop", rmi_b)
```
@weilixu  What do you think?

Reply:
In that case, the RDS should probably be updated to match.

Reply:
Fixed. Thank you for this comment. This is brilliant.